### ISSUE TITLE
Executive can promote himself to Admin

### ISSUE THREAD

**Issue by enginozsoy** [rappasoft/laravel-5-boilerplate#659]
Hi There!

In the user management function a user (ie executive) can grant himself higher roles than himself (ie admin). Maybe it might be useful to have grades of user roles and only make possible to grant roles to others up until the grade of the user himself. 
Version: 4.2.2

Thanks for the excellent work, keep up :)

Regards,
Engin

**Comment by poseso** [comment]
https://github.com/rappasoft/laravel-5-boilerplate/pull/600/commits

**Comment by rappasoft** [comment]
i don't even remember that pull request.

**Comment by poseso** [comment]
I actually use that PR  since 5.3

**Comment by poseso** [comment]
It was open for 2 days but nothing happened so he closed the PR

**Comment by poseso** [comment]
Maybe instead of locking to admins only maybe you can do it for id 1 which is the super user?

**Comment by rienheuver** [comment]
Perhaps it can be so that a user can only upgrade the roles of others to his own level and not above? Because granting yourself admin rights is quite a security issue

**Comment by rappasoft** [comment]
Hows this: https://github.com/rappasoft/laravel-5-boilerplate/commit/279c00327d922aab6e0e68badc756141c429d005

**Comment by mhetreramesh** [comment]
This one is really important feature to have.

**Comment by rappasoft** [comment]
If anyone has ideas feel free to make a PR.

**Comment by Sarahphp1** [comment]
Hi all,
So, I implemented the changes in PR:
https://github.com/rappasoft/laravel-5-boilerplate/pull/600/commits
Just like poseso suggested, and implemented the changes in 279c003 by rappasoft. And I added this code:
`                                @if(!access()->hasRole('Administrator') && $role->id == 1)
                                    @continue
                                @endif`
in /resources/views/backend/access/edit.blade.php after line: 63
In order to prevent user with 'executive' role to promote regular user as administrator in the backend 'edit' view.
And now, everything looks good. 
Probably, I still need to add some code to the 'create' method in:
/app/Repositories/Backend/Access/User/UserRepository.php
I will check it out and let you know!
